src/streams.lisp

```diff
@@ -470,19 +516,32 @@ ALPN-PROTOCOLS, if specified, should be a list of alpn protocol names such as
                  (cipher-list *default-cipher-list*)
                  (buffer-size *default-buffer-size*)
                  (input-buffer-size buffer-size)
-                 (output-buffer-size buffer-size))
+                 (output-buffer-size buffer-size)
+                 (honor-underlying-stream-timeout #+clozure-common-lisp t nil)
+                 (read-timeout nil)
+                 (write-timeout nil))
   "Returns an SSL stream for the server socket descriptor SOCKET.
 CERTIFICATE is the path to a file containing the PEM-encoded certificate for
  your server. KEY is the path to the PEM-encoded key for the server, which
-may be associated with the passphrase PASSWORD."
+may be associated with the passphrase PASSWORD.
+
+READ-TIMEOUT and WRITE-TIMEOUT specify a timeout in seconds before
+which to signal an SSL-TIMEOUT error.  If these are not specified and
+HONOR-UNDERLYING-STREAM-TIMEOUT is true, then the underlying SOCKET
+timeout, if set, will be honored and an SSL-TIMEOUT error will be
+thrown except on CCL where an implementation specific error is thrown
+for backwards compatibility"
```

Comment:
We will have to mention (in the docstrings of the two functions and in the documentation of the timeout slots) that the timeout parameters only work for file descriptor BIOs, and LispBIOs behave differently.